syntax = "proto3";
import "SerializableQueryPlan.proto";
import "SerializableOperator.proto";
import "SerializableDataType.proto";

service WorkerRPCService {
  rpc DeployQuery (DeployQueryRequest) returns (DeployQueryReply) {}
  rpc UndeployQuery (UndeployQueryRequest) returns (UndeployQueryReply) {}

  rpc RegisterQuery (RegisterQueryRequest) returns (RegisterQueryReply) {}
  rpc UnregisterQuery (UnregisterQueryRequest) returns (UnregisterQueryReply) {}

  rpc StartQuery (StartQueryRequest) returns (StartQueryReply) {}
  rpc StopQuery (StopQueryRequest) returns (StopQueryReply) {}

  rpc RequestMonitoringData (MonitoringRequest) returns (MonitoringReply) {}
}

message DeployQueryRequest {
  NES.SerializableQueryPlan queryPlan = 3;
}

message DeployQueryReply {
  bool success = 1;
}

message UndeployQueryRequest {
  uint64 queryId = 1;
}

message UndeployQueryReply {
  bool success = 1;
}

message RegisterQueryRequest {
  NES.SerializableQueryPlan queryPlan = 3;

}

message RegisterQueryReply {
  bool success = 1;
}

message UnregisterQueryRequest {
  uint64 queryId = 1;
}

message UnregisterQueryReply {
  bool success = 1;
}

message StartQueryRequest {
  uint64 queryId = 1;
}

message StartQueryReply {
  bool success = 1;
}

message StopQueryRequest {
  uint64 queryId = 1;
}

message StopQueryReply {
  bool success = 1;
}

message MonitoringRequest {
  NES.SerializableMonitoringPlan monitoringPlan = 1;
}

message MonitoringReply {
  NES.SerializableSchema schema = 1;
  bytes buffer = 2;
}